pkg/metrics/exporter.go

```diff
@@ -0,0 +1,61 @@
+package metrics
+
+import (
+	"flag"
+	"fmt"
+	"strings"
+	"sync"
+
+	"go.opencensus.io/stats/view"
+)
+
+var (
+	curMetricsExporter view.Exporter
+	metricsMux         sync.RWMutex
+)
+
+var (
+	metricsBackend = flag.String("metricsBackend", "Prometheus", "Backend used for metrics")
+	prometheusPort = flag.Int("prometheusPort", 9090, "Prometheus port")
+)
+
+const prometheusExporter = "prometheus"
+
+func NewMetricsExporter() (view.Exporter, error) {
+	ce := getCurMetricsExporter()
+	// If there is a Prometheus Exporter server running, stop it.
+	resetCurPromSrv()
+
+	if ce != nil {
+		// UnregisterExporter is idempotent and it can be called multiple times for the same exporter
+		// without side effects.
+		view.UnregisterExporter(ce)
+	}
+	var e view.Exporter
+	var err error
+	m := strings.ToLower(*metricsBackend)
+	switch m {
+	// Prometheus is the only exporter for now
+	case prometheusExporter:
+		e, err = newPrometheusExporter()
+	default:
+		err = fmt.Errorf("unsupported metrics backend %v", *metricsBackend)
+	}
+	if err != nil {
+		return nil, err
+	}
+	return e, nil
+}
+
+func getCurMetricsExporter() view.Exporter {
+	metricsMux.RLock()
+	defer metricsMux.RUnlock()
+	return curMetricsExporter
+}
+
+func SetCurMetricsExporter(e view.Exporter) {
```

Comment:
updated. let me know what you think

Reply:
LGTM. I'm on the fence between GetMetricsExporter() and NewMetricsExporter() as the behavior changes depending on when this is called. Maybe just MetricsExporter()?

Reply:
NVM, NewMetricsExporter() always creates a new one. The name is apt.

Reply:
Style nit picks: avoid "Get" as a prefix, per https://golang.org/doc/effective_go.html#Getters . Also avoid package stutter, per https://golang.org/doc/effective_go.html#package-names
With those in mind, I suggest New() so that you have `exporter.New()`. This is coding style, so feel free defer for now.